### PR TITLE
Add warnings when using large coefficients

### PR DESCRIPTION
**Pull request by odow**
As [stated by Gurobi](https://groups.google.com/forum/#!topic/gurobi/M_Bzc6RWMhU), "everything that is larger than 1e+100 is silently converted into 1e+100. For bounds, the value is actually 1e+30. So, every bound value that is larger than 1e+30 (smaller than -1e+30) is truncated to 1e+30 (-1e+30)."

This pull requests introduces a warning rather than silently failing. It is made slightly more complicated by the fact we have to propagate the truncation through the `m.obj` vector we keep around to verify Gurobi stores the correct objective.

Closes https://github.com/JuliaOpt/Gurobi.jl/issues/94.

### CODE REVIEW

**Comment by odow** [comment]
cc @blegat @mburq

File: src/GurobiSolverInterface.jl [hunk 1]
```diff
@@ -104,6 +104,7 @@ end
 function loadproblem!(m::GurobiMathProgModel, filename::AbstractString)
     read_model(m.inner, filename)
     m.obj = getobj(m)
+    _truncateobj!(m.obj)
```
Comment (blegat) [inline src/GurobiSolverInterface.jl]: Why don't you put any warnings here ?

File: src/GurobiSolverInterface.jl [hunk 2]
```diff
@@ -155,10 +156,24 @@ function loadproblem!(m::GurobiMathProgModel, A, collb, colub, obj, rowlb, rowub
 
   m.lb, m.ub = rowlb, rowub
   m.obj = copy(obj)
+  _truncateobj!(m.obj)
```
Comment (blegat) [inline src/GurobiSolverInterface.jl]: or here

File: src/GurobiSolverInterface.jl [hunk 3]
```diff
@@ -201,14 +216,18 @@ setconstrLB!(m::GurobiMathProgModel, lb) = (m.changed_constr_bounds = true; m.lb
 setconstrUB!(m::GurobiMathProgModel, ub) = (m.changed_constr_bounds = true; m.ub = copy(ub))
 
 getobj(m::GurobiMathProgModel)     = get_dblattrarray( m.inner, "Obj", 1, num_vars(m.inner)   )
-setobj!(m::GurobiMathProgModel, c) = (m.obj=copy(c); set_dblattrarray!(m.inner, "Obj", 1, num_vars(m.inner), c))
+function setobj!(m::GurobiMathProgModel, c)
+    m.obj = copy(c)
+    _truncateobj!(m.obj)
```
Comment (blegat) [inline src/GurobiSolverInterface.jl]: or here

File: src/GurobiSolverInterface.jl (hunk 2 above)
Comment (odow) [inline src/GurobiSolverInterface.jl]: should get called in the `add_cvars!` function.

**Comment by mlubin** [comment]
This now has a conflict

**Comment by odow** [comment]
@mlubin resolved.

**Pull request merged**